src/dpp/voice/enabled/handle_frame.cpp

```diff
@@ -0,0 +1,450 @@
+/************************************************************************************
+ *
+ * D++, A Lightweight C++ library for Discord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2021 Craig Edwards and D++ contributors
+ * (https://github.com/brainboxdotcc/DPP/graphs/contributors)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ************************************************************************************/
+
+#include <string_view>
+#include <dpp/exception.h>
+#include <dpp/isa_detection.h>
+#include <dpp/discordvoiceclient.h>
+#include <dpp/json.h>
+#include "../../dave/encryptor.h"
+
+#include "enabled.h"
+
+namespace dpp {
+
+bool discord_voice_client::handle_frame(const std::string &data, ws_opcode opcode) {
+	json j;
+
+	/**
+	 * MLS frames come in as type OP_BINARY, we can also reply to them as type OP_BINARY.
+	 */
+	if (opcode == OP_BINARY && data.size() >= sizeof(dave_binary_header_t)) {
+
+		auto* dave_header = reinterpret_cast<const dave_binary_header_t*>(data.data());
```

Comment:
This is UB on std::string, as `char*` cannot act as storage for objects anymore. Can we use a vector of `std::byte`, `unsigned char`, or the worst but the one we use often anyways, `uint8_t`?

Reply:
but uint8_t is unsigned char?
![image](https://github.com/user-attachments/assets/c06e6ffa-f135-46bb-99c7-111029d59d27)

Reply:
Not required to be, but it'll work on most platforms.